### PR TITLE
Add few proxy throughput logs

### PR DESCRIPTION
**Pull request by Viv-Rajkumar**
- Proxy throughput per sec
- Total/Current per client throughput amounts

### CODE REVIEW

File: src/peer_manager.rs [hunk 1]
```diff
@@ -962,14 +967,40 @@ impl PeerManager {
...
+            .get_mut(pub_id)
+            .map(|peer| if let PeerState::Client {
+                            ip,
+                            traffic: old_traffic,
+                        } = *peer.state() {
+                     let new_traffic = old_traffic + added_bytes;
+                     let log_mult = 10 * 1024 * 1024;
+                     let old_rem = old_traffic % log_mult;
+                     let new_rem = new_traffic % log_mult;
+                     if new_rem < old_rem || added_bytes / log_mult > 0 {
```
Comment (maqi) [inline src/peer_manager.rs]: will `if new_traffic % log_mult < added_bytes {}` do the same job?

File: src/peer_manager.rs [hunk 2]
```diff
@@ -962,14 +967,40 @@ impl PeerManager {
...
+    /// Updates the given clients total traffic amount.
+    pub fn add_client_traffic(&mut self, pub_id: &PublicId, added_bytes: u64) {
+        let self_pfx = format!("{:?}", self);
+        let _ = self.peers
+            .get_mut(pub_id)
+            .map(|peer| if let PeerState::Client {
+                            ip,
+                            traffic: old_traffic,
+                        } = *peer.state() {
+                     let new_traffic = old_traffic + added_bytes;
```
Comment (maqi) [inline src/peer_manager.rs]: i know the chance will be very tiny as we are using u64 for the traffic.
however, shall we still consider using `wrapping_add` ?

File: src/peer_manager.rs [hunk 3]
```diff
@@ -157,7 +157,12 @@ pub enum PeerState {
     /// We are connected - via a tunnel if the field is `true`.
     Connected(bool),
     /// We are the proxy for the client
-    Client(IpAddr),
+    Client {
+        /// Client IP
+        ip: IpAddr,
+        /// Traffic relayed for Client
```
Comment (maqi) [inline src/peer_manager.rs]: it's actually the `charge` that rate_limiter charges, for get requests, client will be over charged (even consider the traffic of response)
for put requests, the response traffic is not counted in.

maybe the comment can be modified to make it clear?

**Comment by Viv-Rajkumar** [comment]
Closing this for now. Should come in as part of the whitelist removal PR later on

**Pull request closed without merging**